pkg/policy/snapshots.go

```diff
@@ -0,0 +1,86 @@
+package policy
+
+import (
+	"sync"
+
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+)
+
+const (
+	maxSnapshots = MaxPolicies
+)
+
+type snapshot struct {
+	version  uint32
+	policies *Policies
+}
+
+type Snapshots struct {
+	murw        sync.RWMutex
+	lastVersion uint32
+	snaps       [maxSnapshots]*snapshot
+	nextIdx     int
+	storedCnt   int
+}
+
+var snapshots = &Snapshots{
+	murw:        sync.RWMutex{},
+	lastVersion: 0,
+	snaps:       [maxSnapshots]*snapshot{},
+	nextIdx:     0,
+	storedCnt:   0,
+}
+
+func (ps *Policies) StoreSnapshot() error {
+	snapshots.murw.Lock()
+	defer snapshots.murw.Unlock()
+
+	snapshots.lastVersion++ // new version
+
+	ps.rwmu.Lock()
+	ps.version = snapshots.lastVersion
+	ps.rwmu.Unlock()
```

Comment:
Here you protect version with rwmu but in policies.go you do:
```
func (ps *Policies) Version() uint32 {
	return atomic.LoadUint32(&ps.version)
}
```
I would say you should protect it in one way or the other, and maybe you can use atomic integer instead of these protections

Reply:
Indeed. I'm going to use atomic for that.

Reply:
Fixed.